silx/gui/plot/ProfileMainWindow.py

```diff
@@ -68,41 +73,68 @@ def __init__(self, parent=None, backend=None):
         self._backend = backend
         # by default, profile is assumed to be a 1D curve
         self._profileType = None
+
+        widget = qt.QWidget()
+        self._layout = qt.QStackedLayout(widget)
+        self._layout.setContentsMargins(0, 0, 0, 0)
+        self.setCentralWidget(widget)
+
         self.setProfileType("1D")
         self.setProfileMethod('sum')
 
+    def _getPlot1D(self, init=True):
+        if not init:
+            return self._plot1D
+        if self._plot1D is None:
+            # import here to avoid circular import
+            from .PlotWindow import Plot1D
+            self._plot1D = Plot1D(parent=self, backend=self._backend)
+            self._plot1D.setDataMargins(yMinMargin=0.1, yMaxMargin=0.1)
+            self._plot1D.setGraphYLabel('Profile')
+            self._plot1D.setGraphXLabel('')
+            self._layout.addWidget(self._plot1D)
+        return self._plot1D
+
+    def _showPlot1D(self):
+        plot = self._getPlot1D()
+        self._layout.setCurrentWidget(plot)
+
+    def _getPlot2D(self, init=True):
+        if not init:
+            return self._plot1D
+        if self._plot2D is None:
+            # import here to avoid circular import
+            from .PlotWindow import Plot2D
+            self._plot2D = Plot2D(parent=self, backend=self._backend)
+            self._layout.addWidget(self._plot2D)
+        return self._plot2D
+
+    def _showPlot2D(self):
+        plot = self._getPlot2D()
+        self._layout.setCurrentWidget(plot)
+
     def setProfileType(self, profileType):
         """Set which profile plot widget (1D or 2D) is to be used
 
+        Note: This method should be removed.
```

Comment:
deprecate?

Reply:
Maybe we can talk about it in #3011 cause i reworked everything there

Reply:
I just checked, i put a deprecation in the next PR.

Let me k now, if you prefer, i can remove this comment from this PR.

Reply:
If you've done it in the next PR it's fine.